### DISCUSSION TITLE
Adding Extra Space to JointJS Paper Without PaperScroller

### DISCUSSION THREAD

**Discussion by VADDIJASWANT** [clientIO/joint#2732] [Q&A] [answered]
### Introduction

Hello everyone,

I am working on a project using JointJS where I need to add extra space to the top or left of a paper. Currently, I'm using a standard dia.Paper without PaperScroller, and I'm encountering a limitation where the edges of the paper restrict my ability to add more space.

I have considered a couple of solutions:

Using PaperScroller: This approach would allow for an infinite paper experience, but due to some constraints in my project, I am unable to implement it.
Using paper.translate(): Translating the paper could provide the extra space I need, but this changes the origin of the paper, which could potentially complicate positioning and interactions.
I’m looking for alternative methods or best practices to achieve the following:

Add additional space to the top or left of the paper: Without using PaperScroller, is there a way to dynamically add more space or adjust the visible area?
Avoid changing the paper origin: If there’s a way to manage extra space without impacting the current origin, that would be ideal.
Has anyone faced a similar issue or have suggestions for managing paper space in JointJS without relying on PaperScroller? Any insights or recommendations would be greatly appreciated.

Thank you!

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
You have 2 options:
- move the paper origin: https://docs.jointjs.com/api/dia/Paper/#translate
- move the graph content: https://docs.jointjs.com/api/dia/Graph/#translate

I don't fully understand why you consider moving the paper origin as an issue. Especially if you use the local "graph" coordinates in your application, nothing will change.
